### PR TITLE
Simplifying `JNLPLauncher`

### PR DESCRIPTION
**Pull request by jglick**
https://github.com/jenkinsci/jenkins/pull/8639#discussion_r1372124881 cleaning up from https://github.com/jenkinsci/remoting/pull/677. The situation is more complicated than that in #6543 because the deprecated properties _will_ still be used if you use the deprecated `-jnlpUrl` mode.

### Testing done

https://github.com/jenkinsci/configuration-as-code-plugin/pull/2430 continues to pass against this change. Interactively:

#### Preferred config
inbound-config

#### Deprecated config
When reconfiguring an agent predating this change.
inbound-config-deprecated

#### Launcher page
inbound-main

### Proposed changelog entries

- Deprecating all configurable options in **Launch agent by connecting it to the controller** (`inbound` in JCasC), as these are only useful in conjunction with the deprecated `-jnlpUrl` mode.

### Proposed upgrade guidelines

Reset all inbound launcher options to defaults, and follow new guidance for command-line options for `agent.jar`.

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly
Comment (basil) [inline core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly:85]: Does not seem to work.

```suggestion
          <j:if test="${!it.launcher.configured}">
```

File: core/src/main/java/hudson/slaves/JNLPLauncher.java
Comment (basil) [inline core/src/main/java/hudson/slaves/JNLPLauncher.java:116]: I could understand leaving behind the mess in `JNLPLauncherTest`, but these two usages in this repository seem trivial enough to migrate:

```java
diff --git a/core/src/main/java/hudson/model/Slave.java b/core/src/main/java/hudson/model/Slave.java
index 5077523789..58f99d4598 100644
--- a/core/src/main/java/hudson/model/Slave.java
+++ b/core/src/main/java/hudson/model/Slave.java
@@ -246,7 +246,7 @@ public abstract class Slave extends Node implements Serializable {
             }
         }
         // Default launcher does not use Work Directory
-        return launcher == null ? new JNLPLauncher(false) : launcher;
+        return launcher == null ? new JNLPLauncher() : launcher;
     }
 
     public void setLauncher(ComputerLauncher launcher) {
diff --git a/test/src/test/java/hudson/tools/InstallerTranslatorTest.java b/test/src/test/java/hudson/tools/InstallerTranslatorTest.java
index 2e9fac2e61..1f82bcbe5b 100644
--- a/test/src/test/java/hudson/tools/InstallerTranslatorTest.java
+++ b/test/src/test/java/hudson/tools/InstallerTranslatorTest.java
@@ -52,7 +52,7 @@ public class InstallerTranslatorTest {
 
     @Issue("JENKINS-23517")
     @Test public void offlineNodeForJDK() throws Exception {
-        Node slave = new DumbSlave("disconnected-slave", null, "/wherever", "1", Node.Mode.NORMAL, null, new JNLPLauncher(true), RetentionStrategy.NOOP, Collections.emptyList());
+        Node slave = new DumbSlave("disconnected-slave", null, "/wherever", "1", Node.Mode.NORMAL, null, new JNLPLauncher(), RetentionStrategy.NOOP, Collections.emptyList());
         String globalDefaultLocation = "/usr/lib/jdk";
         JDK jdk = new JDK("my-jdk", globalDefaultLocation, List.of(new InstallSourceProperty(List.of(new CommandInstaller(null, "irrelevant", "/opt/jdk")))));
         r.jenkins.getJDKs().add(jdk);
```

**Review by basil — APPROVED** [review]
Very nice!

File: core/src/main/java/hudson/slaves/JNLPLauncher.java
Comment (jglick) [reply core/src/main/java/hudson/slaves/JNLPLauncher.java:116]: Agreed, those usages should be updated.

File: core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly
Comment (jglick) [inline core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly:94]: inbound-instance-identity

File: core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly
Comment (jglick) [inline core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly:88]: Similar to older message, but not marked as an error since it may be perfectly normal in a WebSockets-only installation.

File: core/src/main/resources/hudson/slaves/JNLPLauncher/main_es.properties
Comment (jglick) [inline core/src/main/resources/hudson/slaves/JNLPLauncher/main_es.properties:23]: Renaming key and deleting old translations since there is now a significant extra clause about WebSockets.

**Review by basil — APPROVED** [review]

**Comment by jglick** [comment]
OOME creating the WAR; probably unrelated?

**Review by NotMyFault — APPROVED** [review]

**Comment by NotMyFault** [comment]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
